File: LeetCode/Easy/0160-intersection-of-two-linked-lists/0160-intersection-of-two-linked-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
        HashSet<ListNode> exist = new HashSet<>();
        
        while (headA != null) {
            exist.add(headA);
            headA = headA.next;
        }
        
        while (headB != null) {
            if (exist.contains(headB)){
                return headB;
            }
            headB = headB.next;
        }
        
        return null;
    }
}

File: LeetCode/Easy/0160-intersection-of-two-linked-lists/0160-intersection-of-two-linked-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode *getIntersectionNode(ListNode *headA, ListNode *headB) {
        set<ListNode*> exist;
        
        while (headA) {
            exist.insert(headA);
            headA = headA->next;
        }
        
        while (headB) {
            if (exist.find(headB) != exist.end()){
                return headB;
            }
            headB = headB->next;
        }
        
        return NULL;
    }
};